code/modules/chemistry/Reagents-Medical.dm

```diff
@@ -146,20 +149,60 @@ datum
 					REMOVE_ATOM_PROPERTY(M, PROP_MOB_STAMINA_REGEN_BONUS, "r_ether")
 				..()
 
+			proc/ether_fireflash(var/volume) // Proc for all of the fireflash reactions
+				if(!temp_reacted)
+					temp_reacted = 1
+					var/radius = clamp(volume*0.25, 0, 3) // Even the smoke might make a big problem here
+					var/list/covered = holder.covered_turf()
+					for(var/turf/t in covered)
+						radius = clamp((volume/covered.len)*0.25, 0, 5)
+						fireflash_s(t, radius, rand(2000, 3000), 500)
+				holder?.del_reagent(id)
+
+			reaction_temperature(exposed_temperature, exposed_volume)
+				if (ismob(holder?.my_atom) && volume < 50) // We don't want this stuff exploding inside people..
+					return
+				ether_fireflash(volume)
+				return
+
+			reaction_obj(var/obj/O, var/volume)
+				if (isnull(O)) return
+				if(isitem(O))
+					var/obj/item/I = O
+					if(I.firesource) // Direct contact with any firesource is enough to cause the ether to combust
+						ether_fireflash(volume)
+				return
+
+			reaction_mob(var/mob/M, var/method=TOUCH, var/volume)
+				. = ..()
+				if(method == TOUCH)
+					var/mob/living/L = M
+					if(istype(L) && L.getStatusDuration("burning"))
+						ether_fireflash(volume)
+				return
+
 			on_mob_life(var/mob/M, var/mult = 1)
 				if(!M) M = holder.my_atom
 				if(!counter) counter = 1
 				M.jitteriness = max(M.jitteriness-25,0)
 				if(M.hasStatus("stimulants"))
-					M.changeStatus("stimulants", -7.5 SECONDS * mult)
+					M.changeStatus("stimulants", -4 SECONDS * mult)
+				if(M.hasStatus("recent_trauma")) // can be used to help fix recent trauma
+					M.changeStatus("recent_trauma", -2 SECONDS * mult)
+				if(holder.has_reagent(src.id,10)) // large doses progress somewhat faster than small ones
```

Comment:
The PR description says that it has increased depletion rate while the mob is KOd, that isn't here. Not sure we want a 0.2 depletion rate knockout chemical, especially with the offgassing behaviour potentially leading to easy accidental perma-stuns.

Reply:
Oh, it used to, but since the amount needed for a KO is 8, i wasn't sure if the on KO change was necessary, i can definitely re-add the on KO change if continuous new doses are a problem. I personally don't think they would be considering the offgass happens in very minute amounts, though.